linking errors using ubuntu-latest image

```
2019-10-28T21:29:00.1338506Z ========== Building cecs ==========
2019-10-28T21:29:00.1340975Z gcc src/systems/gravity.o src/cecs_component.o src/cecs.o src/cecs_system.o src/cecs_entity.o src/cecs_err.o src/main.o -g -Wall -Isystems/ -Icomponents/ -Isrc/ -lm -D_REENTRANT -std=c11 -o cecs
2019-10-28T21:29:00.3957207Z gcc -g -Wall -Isystems/ -Icomponents/ -Isrc/   -c -o tests/runner.o tests/runner.c
2019-10-28T21:29:00.4269640Z gcc -g -Wall -Isystems/ -Icomponents/ -Isrc/   -c -o tests/check_main.o tests/check_main.c
2019-10-28T21:29:00.4591288Z gcc -g -Wall -Isystems/ -Icomponents/ -Isrc/   -c -o tests/arr_check.o tests/arr_check.c
2019-10-28T21:29:00.4801876Z tests/arr_check.c: In function â€˜arr_initâ€™:
2019-10-28T21:29:00.4802839Z tests/arr_check.c:15:2: warning: implicit declaration of function â€˜ck_assert_ptr_nullâ€™; did you mean â€˜ck_assert_ptr_neâ€™? [-Wimplicit-function-declaration]
2019-10-28T21:29:00.4803036Z   ck_assert_ptr_null(arr.data);
2019-10-28T21:29:00.4803110Z   ^~~~~~~~~~~~~~~~~~
2019-10-28T21:29:00.4803215Z   ck_assert_ptr_ne
2019-10-28T21:29:00.4816611Z In file included from tests/arr_check.c:5:0:
2019-10-28T21:29:00.4817335Z tests/arr_check.c: In function â€˜arr_pushâ€™:
2019-10-28T21:29:00.4817839Z src/arr.h:44:16: warning: implicit declaration of function â€˜reallocarrayâ€™; did you mean â€˜reallocâ€™? [-Wimplicit-function-declaration]
2019-10-28T21:29:00.4817993Z     void* tmp = reallocarray(array.data,\
2019-10-28T21:29:00.4818121Z     
```

notable snippets from this log statement are ```2019-10-28T21:29:00.4802839Z tests/arr_check.c:15:2: warning: implicit declaration of function â€˜ck_assert_ptr_nullâ€™; did you mean â€˜ck_assert_ptr_neâ€™? [-Wimplicit-function-declaration]``` and ```2019-10-28T21:29:00.4817335Z tests/arr_check.c: In function â€˜arr_pushâ€™:
2019-10-28T21:29:00.4817839Z src/arr.h:44:16: warning: implicit declaration of function```.
These indicate that the ubuntu-latest image is having issues linking against libcheck and stdlib?

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -15,8 +15,13 @@ jobs:
     - uses: actions/checkout@v1
     - name: install dependencies
       run: |
-        sudo apt-get update -y
-        sudo apt-get install check gcc -y
+        git clone git@github.com:libcheck/check.git --branch 0.13.0
+        autoreconf --install
+        ./configure
+        make
+        make check
+        sudo make install
+        sudo ldconfig
     - name: gcc version check
       run: |
         gcc --version

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -24,6 +24,7 @@ jobs:
         sudo make install
         sudo ldconfig
         cd ..
+        rm  -rf check
     - name: gcc version check
       run: |
         gcc --version

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -15,7 +15,7 @@ jobs:
     - uses: actions/checkout@v1
     - name: install dependencies
       run: |
-        git clone git@github.com:libcheck/check.git --branch 0.13.0
+        git clone https://github.com/libcheck/check.git --branch 0.13.0
         autoreconf --install
         ./configure
         make

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -16,12 +16,14 @@ jobs:
     - name: install dependencies
       run: |
         git clone https://github.com/libcheck/check.git --branch 0.13.0
+        cd check
         autoreconf --install
         ./configure
         make
         make check
         sudo make install
         sudo ldconfig
+        cd ..
     - name: gcc version check
       run: |
         gcc --version

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -15,8 +15,8 @@ jobs:
     - uses: actions/checkout@v1
     - name: install dependencies
       run: |
-        sudo apt update
-        sudo apt install check
+        sudo apt-get update -y
+        sudo apt-get install check -y
     - name: make
       run: make
     - name: make check

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -20,7 +20,7 @@ jobs:
     - name: gcc version check
       run: |
         gcc --version
-    - name: make
-      run: make
+    - name: make cecs
+      run: make cecs
     - name: make check
       run: make check

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - master
+      - actions
 
 jobs:
   build:

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -1,4 +1,4 @@
-name: C/C++ CI
+name: test ci
 
 on:
   push:
@@ -16,7 +16,7 @@ jobs:
     - name: install dependencies
       run: |
         sudo apt-get update -y
-        sudo apt-get install check -y
+        sudo apt-get install check gcc -y
     - name: gcc version check
       run: |
         gcc --version

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -17,6 +17,9 @@ jobs:
       run: |
         sudo apt-get update -y
         sudo apt-get install check -y
+    - name: gcc version check
+      run: |
+        gcc --version
     - name: make
       run: make
     - name: make check

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -13,6 +13,10 @@ jobs:
     
     steps:
     - uses: actions/checkout@v1
+    - name: install dependencies
+      run: |
+        sudo apt update
+        sudo apt install check
     - name: make
       run: make
     - name: make check

--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
 TARGET = cecs
 TEST_TARGET = check
-LIBS = -lm -D_REENTRANT -std=c11 -lGL -lGLEW -lSDL2 -lSDL2_image
+LIBS = -lm -D_REENTRANT -std=c11
 
 CC = gcc
-CFLAGS = -g -Wall -I/usr/include/SDL2 -Isystems/ -Icomponents/ -Isrc/
+CFLAGS = -g -Wall -Isystems/ -Icomponents/ -Isrc/
 
 .PHONY: default all clean FORCE
 

--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ $(TARGET): $(OBJECTS) src/main.o
 
 $(TEST_TARGET): $(OBJECTS) $(TEST_OBJECTS) FORCE
 	@echo "========== Building $(TEST_TARGET) =========="
-	$(CC) $(CFLAGS) $(OBJECTS) $(TEST_OBJECTS) $(LIBS) -lcheck -o $@
+	$(CC) $(CFLAGS) $(OBJECTS) $(TEST_OBJECTS) $(LIBS) -o $@ -lcheck
 	@echo "========== RUNNING TESTS =========="
 	./$(TEST_TARGET)
 	@echo ""

--- a/src/arr.h
+++ b/src/arr.h
@@ -4,6 +4,8 @@
 */
 
 #pragma once
+#include <stdlib.h>
+
 #define array(type)\
 	struct {\
 		type* data;\

--- a/src/arr.h
+++ b/src/arr.h
@@ -5,6 +5,7 @@
 
 #pragma once
 #include <stdlib.h>
+#include <malloc.h>
 
 #define array(type)\
 	struct {\

